aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -481,6 +626,118 @@ module aptos_framework::delegation_pool {
 
         // save delegation pool ownership and resource account address (inner stake pool address) on `owner`
         move_to(owner, DelegationPoolOwnership { pool_address });
+
+        // All delegation pool enable partial governace voting by default once the feature flag is enabled.
+        if (features::delegation_pool_partial_governance_voting_enabled()) {
+            enable_partial_governance_voting(pool_address);
+        }
+    }
+
+    /// Enable partial governance voting on a stake pool. The voter of this stake pool will be managed by this module.
+    /// THe existing voter will be replaced. The function is permissionless.
+    public entry fun enable_partial_governance_voting(
+        pool_address: address,
+    ) acquires DelegationPool, GovernanceRecords {
+        assert!(features::delegation_pool_partial_governance_voting_enabled(), error::invalid_state(EDISABLED_FUNCTION));
+        assert_delegation_pool_exists(pool_address);
+        let delegation_pool = borrow_global<DelegationPool>(pool_address);
+        let stake_pool_signer = retrieve_stake_pool_owner(delegation_pool);
+        synchronize_delegation_pool(pool_address);
+        // delegated_voter is managed by the stake pool itself.
+        stake::set_delegated_voter(&stake_pool_signer, signer::address_of(&stake_pool_signer));
+
+        move_to(&stake_pool_signer, GovernanceRecords {
+            votes: smart_table::new(),
+            votes_per_proposal: smart_table::new(),
+            vote_delegation: smart_table::new(),
+            delegated_votes: smart_table::new(),
+            vote_events: account::new_event_handle<VoteEvent>(&stake_pool_signer),
+            create_proposal_events: account::new_event_handle<CreateProposalEvent>(&stake_pool_signer),
+            delegate_voting_power_events: account::new_event_handle<DelegateVotingPowerEvent>(&stake_pool_signer),
+        });
+    }
+
+    public entry fun vote(voter: &signer, pool_address: address, proposal_id: u64, voting_power: u64, should_pass: bool) acquires DelegationPool, GovernanceRecords {
+        assert_partial_governance_voting_enabled(pool_address);
+
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+
+        let voter_address = signer::address_of(voter);
+        let remaining_voting_power = calculate_and_update_remaining_voting_power(pool_address, voter_address, proposal_id);
+        if (voting_power > remaining_voting_power) {
+            voting_power = remaining_voting_power;
+        };
+        assert!(voting_power > 0, error::invalid_argument(ENO_VOTING_POWER));
+
+        let governance_records = borrow_global_mut<GovernanceRecords>(pool_address);
+        let stake_pool_remaining_voting_power = aptos_governance::get_remaining_voting_power(pool_address, proposal_id);
```

Comment:
I'd move this code from lines 675-682 into a separate (may be inline?) function so it's more readable. Furthermore, this can be removed later in the release after we enable delegation voting